components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/token/AbstractJWTGenerator.java

```diff
@@ -216,7 +218,18 @@ public String buildBody(TokenValidationContext validationContext) throws APIMana
                     String claimURI = it.next();
                     String claimVal = standardClaims.get(claimURI);
                     List<String> claimList = new ArrayList<String>();
-                    if (userAttributeSeparator != null && claimVal != null && claimVal.contains(userAttributeSeparator)) {
+                    if (claimVal != null && claimVal.contains("{")) {
+                        ObjectMapper mapper = new ObjectMapper();
+                        try {
+                            Map<String, String> map = mapper.readValue(claimVal, Map.class);
+                            jwtClaimsSetBuilder.claim(claimURI, map);
+                        } catch (IOException e) {
+                            // Exception isn't thrown in order to generate jwt without claim, even if an error is
+                            // occurred during the retrieving claims.
+                            log.error("Error while reading claim values", e);
+                        }
+                    } else if (userAttributeSeparator != null && claimVal != null &&
+                            claimVal.contains(userAttributeSeparator)) {
```

Comment:
claimVal != null can be moved out to a if block outside since its checked in both if and else if

Reply:
Yes possible. claimVal != null has been basically checked to avoid NPE comes with claimVal.contains function

Reply:
Suggestion is not to remove it, but to move it to an outer block since its evaluated twice.

Reply:
Yes. Got it. But if do so, this can affect other conditions.
Ex: "exp".equals(claimURI)" This time, it doesn't care about claimVal. It can be either null or not null.